bindings/java/src/java_module_host.c

```diff
@@ -239,8 +248,41 @@ static MODULE_HANDLE JavaModuleHost_Create(BROKER_HANDLE broker, const void* con
                                                 }
                                                 else
                                                 {
-                                                    jobject jModule_object = NewObjectInternal(result->env, jModule_class, jModule_constructor, 3, (jlong)result, jBroker_object, jModule_configuration);
-                                                    exception = JNIFunc(result->env, ExceptionOccurred);
+                                                    jobject jModule_object;
+                                                    if (emptyConstructor)
+                                                    {
+                                                        jModule_object = NewObjectInternal(result->env, jModule_class, jModule_constructor, 0);
```

Comment:
Don't forget to include requirements for these, and to add the requirements comments to the tests and implementation as well.